File: java/src/test/java/problems1/MedianOfTwoSortedArraysTest.java
package problems1;

import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class MedianOfTwoSortedArraysTest extends SampledTest {
    @Override
    public String getFileName() {
        return "median_of_two_sorted_arrays.json";
    }

    @Test
    public void test() {
        for (MedianOfTwoSortedArrays solution : MedianOfTwoSortedArrays.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.findMedianSortedArrays(
                            sample.input.nums1,
                            sample.input.nums2
                        )
                    ).isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, Double> {
        public static class Input {
            public int[] nums1;
            public int[] nums2;
        }
    }
}

File: typescript/__tests__/problems1_100/median-of-two-sorted-arrays.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/median-of-two-sorted-arrays';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    nums1: number[],
    nums2: number[],
  },
  output: number,
  message: string,
}

describe(
    'MedianOfTwoSortedArrays',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        strictEqual(
                            solution.findMedianSortedArrays(
                                sample.input.nums1,
                                sample.input.nums2,
                            ),
                            sample.output,
                            sample.message,
                        ),
                    ),
                ),
        ),
);
